Userland/Libraries/LibWeb/Layout/FlexFormattingContext.cpp

```diff
@@ -642,7 +642,15 @@ void FlexFormattingContext::determine_flex_base_size_and_hypothetical_main_size(
     //         - using stretch-fit main size if the flex basis is indefinite and there is no cross size to resolve the ratio against.
     //         - in response to cross size min/max constraints.
     if (item.box->has_natural_aspect_ratio()) {
-        if (!item.used_flex_basis_is_definite && !has_definite_cross_size(item)) {
+        auto has_intrinsic_dimensions = false;
+        if (item.box->is_replaced_box()) {
+            auto box = dynamic_cast<ReplacedBox*>(item.box.ptr());
+            if (box->has_intrinsic_height() || box->has_intrinsic_width()) {
+                has_intrinsic_dimensions = true;
+            }
+        }
```

Comment:
I'm not exactly sure that this is the best place for this code. I would like some advice on a better place to put this.